src/index.js

```diff
@@ -4,22 +4,85 @@
  * TOC Options
  *
  * @type {Object}
- * @prop {?String} options.after - tag, class, id
+ * @prop {?String} options.insert - insertion information with position key
+ *   ('after', 'before', 'afterChildren', 'beforeChildren') and selector value
+ *   ('tag', '.class', '#id')
  * @prop {?String} options.title - toc title
  * @prop {?Array[String, String, Boolean]} options.toggle - toggle button text options
  * @prop {?boolean} options.ignoreMissingSelector - don't throw an error if the
  *   selector is not found
  */
 module.exports = function (options = {}) {
   const {
-    after = 'h1',
+    insert = { after: 'h1' },
     title = 'Content',
     ignoreMissingSelector = false,
     toggle // ['show', 'hide', true],
   } = options
 
-  if (typeof after !== 'string') {
-    throw new PostHtmlTocError(`unexpected 'options.after': ${after}`)
+  if (typeof insert !== 'object') {
+    throw new PostHtmlTocError(`unexpected 'options.insert': ${insert}`)
+  }
+
+  const inserts = Object.entries(insert)
+
+  if (inserts.length !== 1) {
+    throw new PostHtmlTocError(`unexpected number of entries in 'options.insert': ${insert}`)
```

Comment:
test?

Reply:
I'll add a test after we resolve the format of the option value.

Reply:
:+1: ccb239c55993bfa2b13c6339ea2c1c4cfd450b5c